src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -117,10 +117,10 @@ object Template {
   def apply(recs: Iterator[SamRecord]): Template = {
     var r1: Option[SamRecord] = None
     var r2: Option[SamRecord] = None
-    val r1Supp = ListBuffer[SamRecord]()
-    val r2Supp = ListBuffer[SamRecord]()
-    val r1Sec  = ListBuffer[SamRecord]()
-    val r2Sec  = ListBuffer[SamRecord]()
+    var r1Supp: List[SamRecord] = Nil
+    var r2Supp: List[SamRecord] = Nil
+    var r1Sec : List[SamRecord] = Nil
+    var r2Sec : List[SamRecord] = Nil
```

Comment:
It is.  I changed GroupReads to use `Template` instead of `(SamRecord, SamRecord)` to deal with the variable number of reads per template.  That slowed it down a tiny bit and this restores _most_ of the performance by not allocating four list-builders per template.